File: ruby/g11.parent/Student.rb
#!/usr/bin/ruby
# class declaration
class Student < Person
  ###### INITIALIZER ######
  def initialize(name, gpa=0.0) # constructor
    setName(name)      # call mutator
    setGPA(gpa)        # call mutator
  end

  ###### ACCESSORS (SETTERS) ######
  def setGPA (gpa=0.0)
    @gpa = gpa         # set instance data
  end

  ###### ACCESSORS (GETTERS) ######
  def getGPA
    @gpa               # return instance data
  end
end

File: python/h11.parent/Student.py
#!/usr/bin/env python
from Person import Person    # include Person.py

# class declaration
class Student(Person):
  ###### INITIALIZER ######
  def __init__(self, name, gpa=0.0):
    self.name = name  # call parent's mutator
    self.gpa = gpa    # call mutator

  ###### PROPERTY DECORATORS ######
  @property
  def gpa (self):
    return self._gpa

  @gpa.setter
  def gpa (self, gpa):
      self._gpa = gpa
